router/src/main/java/io/micronaut/web/router/DefaultRouteBuilder.java

```diff
@@ -399,19 +399,38 @@ protected UriRoute buildRoute(HttpMethod httpMethod, String uri, MethodExecution
         return buildRoute(httpMethod.name(), httpMethod, uri, executableHandle);
     }
 
+    /**
+     * Build a route.
+     *
+     * @param httpMethod The HTTP method
+     * @param uri The URI
+     * @param mediaTypes The media types
+     * @param executableHandle The executable handle
+     *
+     * @return an {@link UriRoute}
+     */
+    protected UriRoute buildRoute(HttpMethod httpMethod, String uri, List<MediaType> mediaTypes, MethodExecutionHandle<Object, Object> executableHandle) {
```

Comment:
Please add `@since`
Probably should target 4.2.

Reply:
done